mypy/checkstrformat.py

```diff
@@ -55,7 +55,7 @@ def __init__(self,
         self.exprchk = exprchk
         self.msg = msg
 
-    def check_str_interpolation(self, str: StrExpr, replacements: Node) -> Type:
+    def check_str_interpolation(self, str: Union[StrExpr, BytesExpr, UnicodeExpr], replacements: Node) -> Type:
         """Check the types of the 'replacements' in a string interpolation
         expression: str % replacements
         """
```

Comment:
Can you add a TODO around here that in Python 3 the bytes format characters are actually more restricted than str/unicode format characters? (a) There are fewer supported characters; (b) %s only supports bytes. (In Python 2 these limitations do not apply.)